.github/scripts/deploy-review-app.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+app="pr-$PR_NUMBER-badges-shields"
+region="ewr"
+org="shields-io"
+
+# Get PR JSON from the API
+# This will fail if $PR_NUMBER is not a valid PR
+pr_json=$(curl --fail "https://api.github.com/repos/badges/shields/pulls/$PR_NUMBER")
+
+# Attempt to apply the PR diff to the target branch
+# This will fail if it does not merge cleanly
+curl --location "$(echo "$pr_json" | jq .diff_url -r)" | git apply
```

Comment:
Just to clarify: This will fail to deploy a review app if there is a merge conflict.

Reply:
I'm having a bit of deja vu so apologies if this has been asked and answered before, but is there any reason we wouldn't want to check out the PR ref directly? I.e.

```terminal
git fetch pull/$PR_NUMBER/head:pr-$PR_NUMBER && git checkout pr-$PR_NUMBER 
```

Reply:
I don't think we've discussed it before.

At this stage, there is a very pragmatic reason why I'm doing it this way: There is no open PR which has these files in the source branch yet. So if I check out the source branch for any existing PR, `fly.toml` won't be on it. Merging the PR source branch into the working branch allowed me to test this workflow on some real PRs.

Once we get to the stage where all the open PRs are based off a master branch that has these files on it, it might make more sense to check out the branch directly. That would also allow us to deploy review apps for PRs where there is a merge conflict.

I think in the first case, being able to use this on existing open PRs is useful.

Reply:
One thing I realised when I tried to make a preview app for https://github.com/badges/shields/pull/9014/ was that applying the diff doesn't work if there are binary files in it.

I've switched to merging the source branch (which solves that, but will still fail if there is a conflict), but still sticking with merging for now for the reason above